### PR TITLE
Fix bug with spikes name and username

### PR DESCRIPTION
**Pull request by 4D42**
The name and username of spikes where not the same as the ROIs. This pull request fix the problem and at the same time create a function to save the spikes (similar to the one for ROIs)

### CODE REVIEW

File: utilities.py [hunk 1]
```diff
@@ -224,25 +224,56 @@ def save_roi_signals(
...
+
+        if attrs_add is not None:
+            attrs = {**attrs, **attrs_add}
+
+        save_signal(signals[i], f, f"{path}/{dataset_name}", attrs)
+
+    save_signal(time_, f, f"{path}/{defs.DoricFile.Dataset.TIME}")
+
+
+def save_spike_signals(
```
Comment (doriclenses) [inline utilities.py:243]: Hmm.. spikes are not necessarily named as ROI 1, ROI 2, etc., so not sure if it's the best way to organize things..
But spikes are always named the same way as the input signals, so I would create the names only once and pass it to the `save_roi_signals()` and `save_signals()` (to save spikes).

File: CaImAn/caiman_main.py [hunk 2]
```diff
@@ -332,7 +327,7 @@ def save_caiman_to_doric(
             print(cm_defs.Messages.SAVE_SPIKES, flush = True)
             spikes_grouppath = f"{vpath}/{cm_defs.DoricFile.Group.SPIKES+operationCount}"
             spikes_datapath  = f"{spikes_grouppath}/{vdataset}"
-            utils.save_signals(S > 0, time_, f, spikes_datapath, names, usernames, range_min=0, range_max=1)
+            utils.save_spike_signals(S > 0, time_, f, spikes_datapath, attrs_add={"RangeMin": 0, "RangeMax": 0, "Unit": "Intensity"}, spike_ids = ids)
```
Comment (doriclenses) [inline CaImAn/caiman_main.py:330]: We should save the `S`, not just 0 and 1

File: MiniAn/minian_main.py [hunk 3]
```diff
@@ -837,7 +832,7 @@ def save_minian_to_doric(
             print(mn_defs.Messages.SAVE_SPIKES, flush=True)
             spikes_grouppath = f"{vpath}/{mn_defs.DoricFile.Group.SPIKES+operationCount}"
             spikes_datapath  = f"{spikes_grouppath}/{vdataset}"
-            utils.save_signals(S.values > 0, time_, f, spikes_datapath, names, usernames, range_min=0, range_max=1)
+            utils.save_spike_signals(S.values > 0, time_, f, spikes_datapath, attrs_add={"RangeMin": 0, "RangeMax": 0, "Unit": "AU"}, spike_ids=A.coords["unit_id"].values)
```
Comment (doriclenses) [inline MiniAn/minian_main.py:835]: we should save `S.values`

**Review by doriclenses — CHANGES_REQUESTED** [review]

File: MiniAn/minian_main.py [hunk 4]
```diff
@@ -784,36 +784,28 @@ def save_minian_to_doric(
...
-                for operation in operations:
-                    operationAttrs = utils.load_attributes(f, f"{vpath}/{operation}")
-                    if utils.merge_params(params_doric, params_source) == operationAttrs:
-                        if(len(operation) == len(mn_defs.DoricFile.Group.ROISIGNALS)):
-                            operationCount = ""
-                        else:
-                            operationCount = operation[-1]
-
-                        break
+        operationCount = utils.operation_count(vpath, f, mn_defs.DoricFile.Group.ROISIGNALS, params_doric, params_source)
```
Comment (4D42) [inline MiniAn/minian_main.py:795]: I also created a function to generate the operation count. I don't know why I didn't do that before

File: MiniAn/minian_main.py [hunk 5]
```diff
@@ -784,36 +784,28 @@ def save_minian_to_doric(
...
     print(mn_defs.Messages.GEN_ROI_NAMES, flush = True)
-    names = []
-    usernames = []
-    for i in A.coords["unit_id"].values:
-        names.append("ROI"+str(i).zfill(4))
-        usernames.append("ROI {0}".format(i))
+    roi_ids           = A.coords["unit_id"].values
+    roi_dataset_names = [defs.DoricFile.Dataset.ROI.format(str(id_).zfill(4)) for id_ in roi_ids]
+    roi_usernames     = [defs.DoricFile.Dataset.ROI.format(id_) for id_ in roi_ids]
+    roi_names         = [defs.DoricFile.Dataset.ROI.format(id_) for id_ in roi_ids]
```
Comment (4D42) [inline MiniAn/minian_main.py:790]: I did something like that and them I feed to value to the functions that save the ROI and spikes. What do you think?

**Review by doriclenses — CHANGES_REQUESTED** [review]
I edited a bit, could you check if it's alright?

**Comment by 4D42** [comment]
@doriclenses I did some fixes and it should work properly.  I just had some strange bug but now it's working....
@kapilseh can you test to see if minian and caiman work without problem before validating the pullrequest? :)

**Comment by kapilseh** [comment]
Tested both caiman & minian -> seems to be working fine.

**Comment by kapilseh** [comment]
@4D42 @doriclenses  maybe I am missing something here-  what do we do with the spike signals saved by minian/caiman. For now, I can not load them ( or they can be loaded ??)

**Review by doriclenses — APPROVED** [review]

**Comment by doriclenses** [comment]
@kapilseh they can be loaded in the Signal View, so you need to make sure to load ROI signals to the Signal View not Image View, and then load the spikes. I would load only one signal and their spikes, because loading takes time (@4D42 is working on this issue).

**Review by kapilseh — APPROVED** [review]

**Pull request merged**